src/hooks/useProject/index.ts

```diff
@@ -0,0 +1,62 @@
+import { useQuery, useMutation } from '@tanstack/react-query';
+import {
+  fetchProject,
+  addProject,
+  deleteProject,
+  fetchProjectById,
+} from './request';
+
+const useProject = (projectId?: string, page?: number, limit?: number) => {
```

Comment:
use object for props here since all of the args are optional using object will allow more flexible call.

Reply:
Like this?
interface IProject {
  projectId?: string;
  page?: number;
  limit?: number;
}
1)
const useProject = ({ projectId, page, limit }: IProject) => {
2)
const useProject = (props: IProject) => {

Which one to do first one or second one or any other approach?